designs/2025-bulk-suppressions-api/README.md

```diff
@@ -0,0 +1,305 @@
+
+```javascript
+import fs from "node:fs";
+import { getCacheFile } from "./eslint-helpers.js";
+import { SuppressionsService } from "../services/suppressions-service.js";
+
+class ESLint {
+    constructor(options = {}) {
+        const processedOptions = processOptions(options);
+
+        // ... existing constructor logic to initialize options, linter, cache, configLoader ...
+
+        const suppressionsFilePath = getCacheFile(
+            processedOptions.suppressionsLocation,
+            processedOptions.cwd,
+            { prefix: "suppressions_" }
+         );
+
+        privateMembers.set(this, {
+            options: processedOptions,
+            linter: /* ... */,
+            lintResultCache: /* ... */,
+            configLoader: /* ... */
+            suppressionsFilePath,
+        });
+    }
+
+    async lintFiles(patterns) {
+        const { options, suppressionsFilePath, lintResultCache, /* ... other needed members */ } = privateMembers.get(this);
+        let suppressionResults = null;
+
+        // Existing lint logic to get initial `results` (LintResult[])
+        const results = /* LintResult[] */
+
+        // Persist the cache to disk before applying suppressions.
+        if (lintResultCache) {
+            lintResultCache.reconcile();
+        }
+
+        const finalResults = results.filter(result => !!result);
+
+         if (!fs.existsSync(suppressionsFilePath)) {
+            return finalResults;
+         }
+
+        const suppressions = new SuppressionsService({
+            filePath: suppressionsFilePath,
+            cwd: options.cwd,
+        });
+
+        return suppressions.applySuppressions(
+            finalResults,
+            await suppressions.load(),
+        );
+    }
+
+    async lintText(code, options = {}) {
+        const {
+            options: eslintOptions, // Renamed to avoid conflict with method options
+            suppressionsFilePath,
+            linter,
+            configLoader
+        } = privateMembers.get(this);
+
+        const { filePath: providedFilePath, warnIgnored, ...otherOptions } = options || {};
+
+        const results = [];
+
+        // --- Existing lintText logic to determine config, etc. ---
+
+        // Do lint.
+        results.push(
+            verifyText({
+                text: code,
+                filePath: resolvedFilename.endsWith("__placeholder__.js")
+                    ? "<text>"
+                    : resolvedFilename,
+                configs,
+                cwd,
+                fix: fixer,
+                allowInlineConfig,
+                ruleFilter,
+                stats,
+                linter,
+            }),
+        );
+
+        if (!fs.existsSync(suppressionsFilePath)) {
+            return processLintReport(this, { results });
+        }
+
+        const suppressions = new SuppressionsService({
+            filePath: suppressionsFilePath,
+            cwd: eslintOptions.cwd,
+        });
```

Comment:
Invoking this on every call to `lintText` will result in a lot of excess file system calls for API consumers that directly use `lintText` instead of `lintFiles`.

Reply:
Yes, that's correct. It is better to create a single instance in the constructor. I will correct it.

Reply:
1. https://github.com/eslint/rfcs/pull/133/commits/fa25623767c24c22f0aafd4bae2236103111e728

Reply:
Yes, it's okay to use private properties. Most of the core was written before private class fields existed so we had to work around it by using a weakmap.